x-pack/plugins/ml/public/application/model_management/delete_models_modal.tsx

```diff
@@ -102,54 +111,90 @@ export const DeleteModelsModal: FC<DeleteModelsModalProps> = ({ models, onClose
         </EuiModalHeaderTitle>
       </EuiModalHeader>
 
-      {modelsWithPipelines.length > 0 ? (
-        <EuiModalBody>
+      <EuiModalBody>
+        {modelsWithPipelines.length > 0 ? (
+          <>
+            <EuiCallOut
+              title={
+                <FormattedMessage
+                  id="xpack.ml.trainedModels.modelsList.deleteModal.pipelinesWarningHeader"
+                  defaultMessage="{modelsCount, plural, one {{modelId} has} other {# models have}} associated pipelines."
+                  values={{
+                    modelsCount: modelsWithPipelines.length,
+                    modelId: modelsWithPipelines[0].model_id,
+                  }}
+                />
+              }
+              color="warning"
+              iconType="warning"
+            >
+              <div>
+                <p>
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.modelsList.deleteModal.warningMessage"
+                    defaultMessage="Deleting the trained model and its associated {pipelinesCount, plural, one {pipeline} other {pipelines}} will permanently remove these resources. Any process configured to send data to the {pipelinesCount, plural, one {pipeline} other {pipelines}} will no longer be able to do so once you delete the {pipelinesCount, plural, one {pipeline} other {pipelines}}. Deleting only the trained model will cause failures in the {pipelinesCount, plural, one {pipeline} other {pipelines}} that {pipelinesCount, plural, one {depends} other {depend}} on the model."
+                    values={{ pipelinesCount }}
+                  />
+                </p>
+                <EuiCheckbox
+                  id={'delete-model-pipelines'}
+                  label={
+                    <FormattedMessage
+                      id="xpack.ml.trainedModels.modelsList.deleteModal.approvePipelinesDeletionLabel"
+                      defaultMessage="Delete {pipelinesCount, plural, one {pipeline} other {pipelines}}"
+                      values={{ pipelinesCount }}
+                    />
+                  }
+                  checked={deletePipelines}
+                  onChange={setDeletePipelines.bind(null, (prev) => !prev)}
+                  data-test-subj="mlModelsDeleteModalDeletePipelinesCheckbox"
+                />
+              </div>
+              <ul>
+                {modelsWithPipelines.flatMap((model) => {
+                  return Object.keys(model.pipelines).map((pipelineId) => (
+                    <li key={pipelineId}>{pipelineId}</li>
+                  ));
+                })}
+              </ul>
+            </EuiCallOut>
+            <EuiSpacer size="m" />
+          </>
+        ) : null}
+
+        {modelsWithInferenceAPIs.length > 0 ? (
```

Comment:
Reading the requirements in the corresponding issue https://github.com/elastic/kibana/issues/175128, do we also need to include the inference services in the warning when attempting to stop a deployment?

<img width="589" alt="image" src="https://github.com/elastic/kibana/assets/7405507/7edb99f8-a6be-46d5-a052-632bfa07cbb8">

Reply:
We decided to prevent the deployments created with inference API from stopping, updated [here](https://github.com/elastic/kibana/pull/175880/commits/e2c3677ff590f63af510b8ae4d76fbdd34ec04f4)

Reply:
Just checking this is the expected behavior - logged in as a different user (ml_poweruser) to the one with which I created the inference service (elastic), and get different behavior when I try and stop the deployment:

<img width="629" alt="image" src="https://github.com/elastic/kibana/assets/7405507/a7aeaa2b-9164-4b58-bd87-6831f7e215af">

Here I get no warning about stopping the deployment created with the inference API. When logged in as `elastic` I do see the warning.

Reply:
Testing with https://github.com/elastic/kibana/pull/175880/commits/4156b03faa88ada2cfc27e3a81e5619b91b0eb57 I now get a warning for a user who did not create the service:

<img width="593" alt="image" src="https://github.com/elastic/kibana/assets/7405507/89bae655-5ed0-4355-ac37-ecc6944dcd9c">